Extend support for deprecated Plugins, Editor, Captcha. For pr 45818 and 45819 (#493)
joomla/Manual: migrations/53-54/new-deprecations.md

- ❌ `registerListeners()` is deprecated in both `Joomla\CMS\Extension\PluginInterface` and `Joomla\CMS\Plugin\CMSPlugin`.
  - ✅ Instead, implement the `SubscriberInterface`. The method is no longer required in this case.

> ### Support for deprecated event API were extended from Joomla 6 to Joomla 7.
> 
> [45818](https://github.com/joomla/joomla-cms/pull/45818) Support for deprecated event API were extended from Joomla 6 to Joomla 7.
> Support for events without event class will be stoped in Joomla 7 instead of Joomla 6.
> 
> ### Support for deprecated Editor and Captcha API were extended from Joomla 6 to Joomla 7.
> [45819](https://github.com/joomla/joomla-cms/pull/45819) Support for deprecated Editor and Captcha API were extended from Joomla 6 to Joomla 7.
> Links for an examples with new API:
>  - [Editors Plugin](/docs/building-extensions/plugins/plugin-examples/editors-plugin/)
>  - [Editors Buttons (XTD) Plugin](/docs/building-extensions/plugins/plugin-examples/editors-xtd-plugin/)
>  - [Captcha Plugin](/docs/building-extensions/plugins/plugin-examples/captcha-plugin/)
> 

Reviewer: ```suggestion
### Support for the deprecated event API was extended from Joomla 6 to Joomla 7.

[45818](https://github.com/joomla/joomla-cms/pull/45818): Support for the deprecated event API was extended from Joomla 6 to Joomla 7.
Support for events without an event class will be stopped in Joomla 7 instead of Joomla 6.

### Support for the deprecated Editor and Captcha APIs was extended from Joomla 6 to Joomla 7.
[45819](https://github.com/joomla/joomla-cms/pull/45819): Support for the deprecated Editor and Captcha APIs was extended from Joomla 6 to Joomla 7.
Example implementations using the new APIs:
 - [Editors Plugin](/docs/building-extensions/plugins/plugin-examples/editors-plugin/)
 - [Editors Buttons (XTD) Plugin](/docs/building-extensions/plugins/plugin-examples/editors-xtd-plugin/)
 - [Captcha Plugin](/docs/building-extensions/plugins/plugin-examples/captcha-plugin/)
```
I've tried to apply the grammar suggestions from the bot in the GUI, but that did not work, so I make a change suggestion for that.
